File: cpp/source/event/input/user/Operate.cpp
#include "event/input/user/Operate.hpp"

#include <sstream>
#include <iterator>

using namespace fm;
using namespace fm::event::input::user;

using namespace com::fleetmgr::interfaces;

Operate::Operate(long _deviceId, std::shared_ptr<std::vector<std::shared_ptr<ChannelRequest>>> _channels) :
    UserEvent(OPERATE),
    deviceId(_deviceId),
    channels(_channels)
{
}

long Operate::getDeviceId() const
{
    return deviceId;
}

const std::vector<std::shared_ptr<ChannelRequest>>& Operate::getChannels() const
{
    return *channels;
}

std::string Operate::toString() const
{
    std::ostringstream oss;
    for (std::shared_ptr<ChannelRequest> c : *channels)
    {
        oss << c->id() << ",";
    }
    return "OPERATE: " + std::to_string(deviceId) + " channels: " + oss.str();
}

File: java/src/main/java/com/fleetmgr/sdk/client/event/input/user/Operate.java
package com.fleetmgr.sdk.client.event.input.user;

import com.fleetmgr.interfaces.ChannelRequest;

import java.util.Collection;

/**
 * Created by: Bartosz Nawrot
 * Date: 18.09.2018
 * Description:
 */
public class Operate extends UserEvent {

    private long deviceId;

    private Collection<ChannelRequest> channels;

    public Operate(long deviceId, Collection<ChannelRequest> channels) {
        super(Type.OPERATE);
        this.deviceId = deviceId;
        this.channels = channels;
    }

    public long getDeviceId() {
        return deviceId;
    }

    public Collection<ChannelRequest> getChannels() {
        return channels;
    }

    @Override
    public String toString() {
        return "Operate{" +
                "deviceId=" + deviceId +
                ", channels=" + channels +
                '}';
    }
}
